modules/labs-react/expandable/lib/ExpandableTarget.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+import {
+  createComponent,
+  ExtractProps,
+  focusRing,
+  hideMouseFocus,
+  styled,
+  StyledType,
+  useModelContext,
+} from '@workday/canvas-kit-react/common';
+
+import {ExpandableModelContext} from './Expandable';
+import {DisclosureModel, DisclosureModelConfig} from '@workday/canvas-kit-react/disclosure';
+import {ExpandableTitle} from './ExpandableTitle';
+import {useExpandableTarget} from './hooks/useExpandableTarget';
+import {Flex} from '@workday/canvas-kit-react/layout';
+import {colors} from '@workday/canvas-kit-react/tokens';
+
+export interface ExpandableTargetProps
+  extends DisclosureModelConfig,
+    ExtractProps<typeof Flex, never> {
+  model?: DisclosureModel;
+  /**
+   * Children of the `Expandable.Target`. Should contain `Target.Title`, an optional `Target.Avatar`
+   * and either `Target.StartIcon` or `Target.EndIcon`. `Target.StartIcon` is meant to be placed before
+   * the `Target.Title` and `Target.EndIcon` should be placed after.
+   */
+  children: React.ReactNode;
+  /**
+   * Specifies the semantic heading level that will wrap the `Expandable.Target`'s button. If 'none' is
+   * selected the no semantic heading will wrap the button.
+   */
+  headingLevel: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'none';
+}
+
+const StyledButton = styled(Flex.as('button'))<StyledType>({
+  cursor: 'pointer',
+  '&:focus': {
+    ...focusRing(),
+  },
+  '&:hover': {
+    background: colors.soap300,
+  },
+  ...hideMouseFocus,
+});
+
+const Heading = styled('h1')<StyledType>({
+  margin: 0,
+});
```

Comment:
I left the heading as a styled component. I haven't quite decided on whether I should define a default for the `headingLevel` prop because I like that it's required.

I debated creating a separate compound component for `Heading`, but I felt that it cluttered the API a bit if I keep the `Expandable.Target` logic the same and could lead to accidental nested headings.

```
<Expandable>
  <Expandable.Heading headingLevel="h1">
    <Expandable.Target headingLevel="none">
      <Expandable.StartIcon />
      <Expandable.Title>Title</Expandable.Title>
    </Expandable.Target>
  </Expandable.Heading>

 <Expandable.Content>Content</Expandable.Content> 
</Expandable>
```